src/main/java/org/alfresco/repo/management/JmxDumpUtil.java

```diff
@@ -164,9 +172,129 @@ private static void printMBeanInfo(MBeanServerConnection connection, ObjectName
+        {
+            input = cleanPasswordFromInputArgument(input, passwordRedactPattern);
+            cleanInputs.add(input);
+        }
+        
+        return cleanInputs.toArray(new String[commandInputs.length]);
+    }
+
+    /**
+     * Removes any characters the word/s provided in redactedInputs
+     * and replaces them with JmxDumpUtil.PROTECTED_VALUE
+     * <p>
+     * Example: 
+     * <p>
+     * Input:   -Ddb.password=alfresco
+     * <p>
+     * Output:  -Ddb.password=********
+     * </p>
+     * 
+     * @param input String
+     * @param redactedInputs String[]
+     * @return password redacted string if input matches a string in redactedInputs, an un-altered string will be returned if it does not match.
+     */
+    static String cleanPasswordFromInputArgument(String input, Pattern redactedInputPattern)
+    {
+        //Replace the whole string with just capture group 1 to remove the desired value and concat the protected value.
+        String output = redactedInputPattern.matcher(input).replaceAll("$1"+PROTECTED_VALUE); //input.replaceAll(regex, "$1"+PROTECTED_VALUE);
+        return output;
+    }
+
+    /**
+     * Creates a regular expression that will select a string that contains one of the values provided in argEndings, proceeding an "=" and defines two capture groups:
+     * <ul>
+     * <li>Group 1: An argEnding that is followed by an "=", including the "=" and all character prior to the argEnding.
+     * <li>Group 2: The characters that follow group 1, to the end of the string or new line.
+     * </ul>
+     * <p>
+     * The argEnding can be the whole Input argument or the common characters proceeding the = sign.
+     * Example argEndings:
+     * <ul>
+     * <li>    -Ddb.password   This will select the values passed as -Ddb.password
+     * <li>    password        This will select any potential values that end in the word password
+     * </ul>
+     * <p>
+     * Example usage: 
+     * <p>
+     * argEndings={"password", "pwd"}
+     * <p>
+     * This will create a regex that will match a string that contains either argEndings. 
+     * The following will be matched by the resulting regex:
+     * <p>
+     * "-Ddb.password=my_password"
+     * <p>
+     * For this example: group 1="-Ddb.password=" group 2="my_password"
+     * 
+     * 
+     * @param argEndings Strings that will end the input argument you wish to select
+     * @return Regex pattern for selecting the characters following the strings passed as argEndings
+     */
+    static String createPasswordFindRegexString(String[] argEndings) throws IllegalArgumentException
+    {
+        if(argEndings.length<1)
+        {
+            IllegalArgumentException e = new IllegalArgumentException("Arguments are required");
+            throw e;
+        }
+
+        StringJoiner argJoiner = new StringJoiner("|");
+
+        for (String argEnding : argEndings) 
+        {
+            argJoiner.add(escapeRegexMetaChars(argEnding)+"=");

```

Comment:
You can add a prefix and a suffix using the joiner :)

Reply:
Using suffix with the joiner would just add it to the end of the final joined string, which would output something like this: `"string1|string2|string3="` 
It needs to be `"string1=|string2=|string3="`

Reply:
Ah ok, makes sense, sorry.